waltz-tools/src/main/java/com/wepay/waltz/tools/cluster/ClusterCli.java

```diff
@@ -287,6 +278,154 @@ protected void processCmd(CommandLine cmd) throws SubCommandFailedException {
+            int serverId = -1;
+
+            PartitionAssignment partitionAssignment = clusterManager.partitionAssignment();
+            for (int sId : partitionAssignment.serverIds()) {
+                for (PartitionInfo partitionInfo : partitionAssignment.partitionsFor(sId)) {
+                    if (partitionInfo.partitionId == partitionId) {
+                        serverId = sId;
+                        break;
+                    }
+                }
+                if (serverId != -1) {
+                    break;
+                }
+            }
+
+            Set<ServerDescriptor> serverDescriptors = clusterManager.serverDescriptors();
+            for (ServerDescriptor serverDescriptor : serverDescriptors) {
+                if (serverDescriptor.serverId == serverId) {
+                    serverEndpoint = serverDescriptor.endpoint;
+                }
+            }
+
+            // Step 1: Partition validation on zk
+            String error = (serverId == -1) ? "Error: Partition " + partitionId + " is not assigned to any server" : "";
+
+            ValidationResult validationResult = new ValidationResult(
+                ValidationResult.ValidationType.PARTITION_ASSIGNMENT_ZK_VALIDITY,
+                ("".equals(error)) ? ValidationResult.Status.SUCCESS : ValidationResult.Status.FAILURE,
+                error);
+            partitionValidationResults.validationResultsMap.put(validationResult.type, validationResult);
+
+            // Step 2: Zk and server partition assignment consistency validation
+            CompletableFuture<Void> consistencyValidationFuture =
+                buildServersZKPartitionAssignmentsConsistencyValidationPerPartition(rpcClient, serverEndpoint,
+                    partitionId, partitionValidationResults);
+
+            // Step 3: Build Server-Storage connectivity validations
+            CompletableFuture<Void> connectivityValidationFuture =
+                buildServerStorageConnectivityValidationPerPartition(rpcClient,
+                serverEndpoint, replicaAssignments, partitionId, partitionValidationResults);
+
+            updateRemainingValidationResults(consistencyValidationFuture, connectivityValidationFuture,
+                partitionsValidationResultList);
+
+            ValidationResult storageConsistencyValidationResult = new ValidationResult(
+                ValidationResult.ValidationType.PARTITION_ASSIGNMENT_ZK_STORAGE_CONSISTENCY,
+                ValidationResult.Status.SUCCESS, "");
+            ValidationResult quorumValidationResult = new ValidationResult(
+                ValidationResult.ValidationType.PARTITION_QUORUM_STATUS,
+                ValidationResult.Status.SUCCESS, "");
+            partitionValidationResults.validationResultsMap.put(storageConsistencyValidationResult.type, storageConsistencyValidationResult);
+            partitionValidationResults.validationResultsMap.put(quorumValidationResult.type, quorumValidationResult);
+
```

Comment:
Added these validationResult for `PARTITION_ASSIGNMENT_ZK_STORAGE_CONSISTENCY` and `PARTITION_QUORUM_STATUS` so that the build doesn't fail. These statements should be replaced with the storage connectivity per partition implementation.